File: DevExpress VCL 23.2.6 Full Source/VCL/ExpressQuantumTreeList/Demos/CBuilder/ScrollbarAnnotationsDemo/ScrollbarAnnotationsDemoData.cpp
//---------------------------------------------------------------------------


#pragma hdrstop

#include "ScrollbarAnnotationsDemoData.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma classgroup "Vcl.Controls.TControl"
#pragma link "dxmdaset"
#pragma resource "*.dfm"
TScrollbarAnnotationsDemoDataDM *ScrollbarAnnotationsDemoDataDM;
//---------------------------------------------------------------------------
__fastcall TScrollbarAnnotationsDemoDataDM::TScrollbarAnnotationsDemoDataDM(TComponent* Owner)
	: TDataModule(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TScrollbarAnnotationsDemoDataDM::DataModuleCreate(TObject *Sender)
{
  mdDepartments->LoadFromBinaryFile("..\\..\\Data\\Departments.dat");
  mdDepartments->Open();
}
//---------------------------------------------------------------------------

File: DevExpress VCL 23.2.6 Full Source/VCL/ExpressQuantumTreeList/Demos/Delphi/ScrollbarAnnotationsDemo/ScrollbarAnnotationsDemoData.pas
unit ScrollbarAnnotationsDemoData;

interface

uses
  SysUtils, Classes, DB, dxmdaset;

type
  TScrollbarAnnotationsDemoDataDM = class(TDataModule)
    mdDepartments: TdxMemData;
    mdDepartmentsID: TAutoIncField;
    mdDepartmentsPARENTID: TIntegerField;
    mdDepartmentsMANAGERID: TIntegerField;
    mdDepartmentsNAME: TStringField;
    mdDepartmentsBUDGET: TFloatField;
    mdDepartmentsLOCATION: TStringField;
    mdDepartmentsPHONE: TStringField;
    mdDepartmentsFAX: TStringField;
    mdDepartmentsEMAIL: TStringField;
    mdDepartmentsVACANCY: TBooleanField;
    dsDepartments: TDataSource;
    procedure DataModuleCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  ScrollbarAnnotationsDemoDataDM: TScrollbarAnnotationsDemoDataDM;

implementation

{%CLASSGROUP 'Vcl.Controls.TControl'}

{$R *.dfm}

procedure TScrollbarAnnotationsDemoDataDM.DataModuleCreate(Sender: TObject);
const
  DataPath = '..\..\Data\';
begin
  mdDepartments.LoadFromBinaryFile(DataPath + 'Departments.dat');
  mdDepartments.Open;
end;

end.
